tools/cross_runner/kernels/conv_nchw_fp16.cpp

```diff
@@ -149,6 +149,42 @@ _GENX_ inline vector<DT_ACCU, INPUT_REG_SIZE> load_input_nchw(SurfaceIndex surfa
 	return ret;
 }
 
+_GENX_ inline vector<DT_ACCU, INPUT_REG_SIZE> load_input_nhwc(SurfaceIndex surface [[type("buffer_t")]], uint32_t input_width, uint32_t input_height, uint32_t input_pad, uint32_t w_offset, int32_t h_offset, uint32_t batch_base_offset_bytes)
+{
+	const uint32_t LINEAR_LOAD_SIZE = 64;
+	const int32_t h_offset_pad = h_offset - int32_t(input_pad);
+	vector<DT_ACCU, INPUT_REG_SIZE> ret(0.0f);
+	vector<DT_ACCU, LINEAR_LOAD_SIZE> load_chunk_accu_dt(0.0f);
+	vector<uint8_t, LINEAR_LOAD_SIZE> predicate(0);
+	vector<int32_t, LINEAR_LOAD_SIZE> offsets;
+
+	#pragma unroll
+	for(int i = 0; i < LINEAR_LOAD_SIZE; i++)
+	{
+		offsets[i] = (i - int32_t(input_pad)) * INPUT_ELEMENT_SIZE_I32;
```

Comment:
This is big loop (64 iterations), and currently it do 2 alu ops: subtract and multiply.
1. Is there a way (if performance gain observed) to initialize precomputed offsets with increasing values and in loop only make subtract?
2. If input_pad would be build option this loop would be removed right? What would be performance impact?

Reply:
let me explore this and get back.